src/Model/Order/OrderTest.php

```diff
@@ -41,45 +42,74 @@ private function getAggregate($amount): Order
                 ]),
             ]
         );
         return $order;
     }
 
-    private function createOrder($amount)
+    public function test_order_amount_cannot_be_negative()
     {
+        $this->expectException(OrderAmountMustNotBeNegative::class);
+        $amount = -50;
         Order::create(
-            OrderId::generate(),
-            BusinessName::fromNative("Jimmy's car rental"),
-            BusinessAddress::fromNative([
+            NonNullOrderId::generate(),
+            NonNullBusinessName::fromNative("Jimmy's car rental"),
+            NonNullBusinessAddress::fromNative([
                 'addressLine1' => "No 3. Exeter Road",
                 'addressLine2' => "",
                 'town' => "Exeter",
                 'county' => "Devon",
                 'postcode' => "EX2 4QE",
                 'countryCode' => "GB",
             ]),
-            ContactPerson::fromNative("Jimmy Neutron"),
-            OrderIsPaid::true(),
-            OrderAmount::fromNative([
+            NonNullContactPerson::fromNative("Jimmy Neutron"),
+            NonNullOrderIsPaid::true(),
+            NonNullOrderAmount::fromNative([
                 "amount" => $amount,
                 "currency" => "gbp",
             ])
         );
     }
 
-    public function test_order_amount_cannot_be_negative()
+    public function test_order_amount_cannot_be_zero()
     {
-        $this->expectException(OrderAmountMustNotBeNegative::class);
-        $this->createOrder(-50);
+        $this->expectException(OrderAmountMustBeGreaterThanZero::class);
+        $amount = 0;
+        Order::create(
+            NonNullOrderId::generate(),
+            NonNullBusinessName::fromNative("Jimmy's car rental"),
+            NonNullBusinessAddress::fromNative([
+                'addressLine1' => "No 3. Exeter Road",
+                'addressLine2' => "",
+                'town' => "Exeter",
+                'county' => "Devon",
+                'postcode' => "EX2 4QE",
+                'countryCode' => "GB",
+            ]),
+            NonNullContactPerson::fromNative("Jimmy Neutron"),
+            NonNullOrderIsPaid::true(),
+            NonNullOrderAmount::fromNative([
+                "amount" => $amount,
+                "currency" => "gbp",
+            ])
+        );
     }
 
-    public function test_order_amount_cannot_be_zero()
+    public function test_order_should_record_the_time_it_was_paid()
     {
-        $this->expectException(OrderAmountMustBeGreaterThanZero::class);
-        $this->createOrder(0);
+        $order = $this->getAggregate([
+            "amount" => "50",
+            "currency" => "gbp",
+        ]);
+        $order->pay();
+        $events = $this->popRecordedEvents($order);
+        /** @var OrderWasPaid $event */
+        $event = $events[0];
+        $this->assertInstanceOf(OrderWasPaid::class, $event);
+        $this->assertTrue($order->getIsPaid()->toNative());
+        $this->assertNotNull($event->createdAt());
```

Comment:
I think this should be testing that the date paid is recorded on the aggregate. e.g. something like `$order->getDatePaid()`.